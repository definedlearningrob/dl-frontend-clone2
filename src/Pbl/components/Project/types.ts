import { PRODUCT_SUBMISSION_STATUS } from '@pbl/resources/enums';
import { TProductSubmissionFile } from '@pbl/graphql/student/queries/projectProducts';
import { TTask } from '@pbl/graphql/user/queries/projectProducts';

import { SUBMISSION_GRADE_STATUS } from '@shared/resources/enums';

export type TProduct = {
  description: string;
  displayName: string;
  id: string;
  name: string;
  rubrics: TRubric[];
  submission?: TSubmission;
  submissionsGradingNeededCount: number;
  hidden?: boolean;
  quickTask?: TTask;
};

export type TRubric = {
  criteriaLabels: TCriteriaLabel[];
  criterias: TCriteria[];
  description: string;
  displayName: string;
  headings: THeading[];
  pointsAvailable?: number;
  id: string;
  name: string;
  hasAlignedStatements?: boolean;
};

type TCriteriaLabel = {
  displayName: string;
  id: string;
  score: number;
};

type TCriteria = {
  id: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
};

type THeading = {
  id: string;
  multiplier: number;
  name: string;
};

export type TCheckInQuestionAnswer = {
  answer: string;
  id: string;
  grade: TCheckInQuestionGrade;
  updatedAt: string;
};

export type TCheckInQuestionGrade = {
  createdAt: string;
  lastGradedBy: {
    firstName: string;
    lastName: string;
  };
  status: SUBMISSION_GRADE_STATUS;
  updatedAt: string;
};

export type TCheckInQuestionOwner = {
  uuid: string;
};

export type TCheckInQuestion = {
  answer: TCheckInQuestionAnswer | null;
  id: string;
  gradingNeededCount?: number;
  question: string;
  step: number;
  owner: TCheckInQuestionOwner;
  questionIndex?: number;
  teamSubmission: TCheckInTeamSubmission | null;
  isHidden?: boolean;
  __typename: 'CheckInQuestion';
};

export type TCheckInGroup = {
  displayName: string;
  id: string;
  name: string;
  questions: TCheckInQuestion[];
  step: number;
  __typename: 'CheckInGroup';
};

export type TValues = {
  question: string;
};

export type TSubmission = {
  id: string;
  files: TFile[];
  grade?: TProductSubmissionGrade;
  status: PRODUCT_SUBMISSION_STATUS;
  canSubmit: boolean;
};

export type TProductSubmission = {
  id: string;
  files: TProductSubmissionFile[];
  grade?: TProductSubmissionGrade;
  status: PRODUCT_SUBMISSION_STATUS;
  canSubmit: boolean;
  updatedAt?: string;
};

export type TProductSubmissionGrade = {
  updatedAt: string;
  lastGradedBy: {
    firstName: string;
    lastName: string;
  };
  pointsAvailable: number;
  pointsScored: number;
  results: TSubmissionGradeResults[];
};

export type TSubmissionGradeResults = {
  criteriaId: string;
  trait?: string;
};

type TFile = {
  createdAt: string;
  filename: string;
  googleWeblink: string | null;
  id: string;
  source: string;
  url: string;
  submitter: {
    firstName: string | null;
    lastName: string | null;
    uuid: string;
  };
};

export type TDefinedCareer = {
  id: string;
  name: string;
  type: string;
  thumbnailUrl: string;
  pathwayName: string;
  __typename: string;
};

export type TCheckInTeamSubmission = {
  id: string;
  grade: TCheckInQuestionGrade;
  answers: {
    answer: string;
    id: string;
    student: {
      uuid: string;
      firstName: string;
      lastName: string;
    };
    updatedAt: string;
  }[];
  canSubmit: boolean;
};
