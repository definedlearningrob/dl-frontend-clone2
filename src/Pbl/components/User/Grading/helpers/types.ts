import { ApolloError, MutationFunctionOptions } from '@apollo/client';

import { GRADING_ITEM_TYPES } from '@pbl/resources/enums';
import { TGradeSubjectsData } from '@pbl/graphql/user/hooks/useGradeSubjectsByStatus';
import {
  TGradeSubmisisonMutationVariables,
  TGradeSubmissionMutationData,
} from '@pbl/graphql/user/hooks/useGradeCheckinMutation';
import { TStudentsAssginedToProjectData } from '@pbl/graphql/user/hooks/useStudentsAssignedToProject';
import { TStudent, TTeam } from '@pbl/graphql/user/queries/gradeSubjectsByStatus';

export type GradeCheckinsOptions = Omit<
  MutationFunctionOptions<TGradeSubmissionMutationData, TGradeSubmisisonMutationVariables>,
  'variables'
>;

export type GradingContextType = {
  navigation: {
    itemId: string;
    isTeamGrading?: boolean;
    classId?: string;
    subjectId?: string;
    hasNextStudent: boolean;
    hasPreviousStudent: boolean;
    projectId: string;
    pickedStudent?: PickedStudent['name'];
    navigateTo: (
      type: TTeam['__typename'] | TStudent['__typename'],
      config: {
        classId: string;
        subjectId: string;
        name: string;
      }
    ) => void;
    goToStudent: (type: 'next' | 'previous') => void;
    openSection: (classId: string) => void;
  };
  gradingType: GRADING_ITEM_TYPES;
  gradeItem: () => void;
  localGradedSubjects: Set<string>;
  studentsList: {
    data: TGradeSubjectsData | TStudentsAssginedToProjectData | undefined;
    loading: boolean;
    error: ApolloError | undefined;
  };
};

export type PickedStudent = {
  uuid: string;
  name: string;
};
