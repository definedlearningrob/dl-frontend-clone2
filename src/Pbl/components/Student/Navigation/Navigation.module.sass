@import '@shared/stylesheets/config.sass'

.navigation
  flex: 1 1 auto
  margin-bottom: auto
