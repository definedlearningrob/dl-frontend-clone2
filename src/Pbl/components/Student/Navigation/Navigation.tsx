import { useTranslation } from 'react-i18next';

import useUserInfo from '@pbl/hooks/useUserInfo';
import { ReactComponent as DashboardIcon } from '@pbl/svg/home_dashboard.svg';
import { TStudentInfo } from '@pbl/graphql/student/queries/userInfo';

import { SidebarNavItem } from '@shared/components/Sidebar/NavItem/NavItem';
import { QuestionBox } from '@shared/components/QuestionBox/QuestionBox';
import { ReactComponent as MessageIcon } from '@shared/svg/messages.svg';
import { ReactComponent as PortfolioIcon } from '@shared/svg/portfolio.svg';
import { ReactComponent as PlansIcon } from '@shared/svg/certificate.svg';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';

import styles from './Navigation.module.sass';

function StudentNavigation() {
  const { t } = useTranslation();
  const { userInfo } = useUserInfo<TStudentInfo>();
  const { isExpanded } = useNavigation();

  const navItems = [
    {
      id: 'dashboard',
      icon: <DashboardIcon />,
      path: '/',
      text: t('student.navigation.dashboard'),
    },
    {
      id: 'portfolio',
      icon: <PortfolioIcon />,
      path: '/portfolio',
      text: t('student.navigation.portfolio'),
    },
    {
      id: 'plans',
      icon: <PlansIcon />,
      path: '/plans',
      text: t('student.navigation.plans'),
      hidden: !userInfo.hasPlans,
    },
    {
      id: 'messaging',
      icon: <MessageIcon />,
      path: '/messages',
      needsAttention: userInfo?.hasUnreadConversation,
      text: t('navigation.messages'),
    },
  ];

  const renderNavItems = navItems.map((item) => <SidebarNavItem key={item.id} {...item} />);

  return (
    <>
      <nav className={styles.navigation}>{renderNavItems}</nav>
      <QuestionBox isExpanded={isExpanded} />
    </>
  );
}

export default StudentNavigation;
