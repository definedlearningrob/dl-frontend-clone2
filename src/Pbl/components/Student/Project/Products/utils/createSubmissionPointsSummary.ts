export const createSubmissionPointsSummary = (pointsAvailable: number, pointsScored: number) =>
  `${pointsScored}/${pointsAvailable}`;
