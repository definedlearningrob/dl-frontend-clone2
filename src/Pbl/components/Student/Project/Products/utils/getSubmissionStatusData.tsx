import { capitalize } from 'lodash-es';

import { PRODUCT_SUBMISSION_STATUS } from '@pbl/resources/enums';

import { SUBMISSION_STATUS } from '@shared/components/SubmissionStatusBadge';

export const getSubmissionStatusData = (
  status: PRODUCT_SUBMISSION_STATUS,
  gradePoints?: string | null
) => {
  const statusLabel = capitalize(status).replace(/[_]/g, ' ');
  const label = [statusLabel, gradePoints].filter(Boolean).join(' ');

  switch (status) {
    case PRODUCT_SUBMISSION_STATUS.DRAFT:
      return { label, value: SUBMISSION_STATUS.DRAFT };
    case PRODUCT_SUBMISSION_STATUS.SUBMITTED:
      return {
        label,
        value: SUBMISSION_STATUS.SUBMITTED,
      };
    case PRODUCT_SUBMISSION_STATUS.GRADED:
      return { label, value: SUBMISSION_STATUS.GRADED };
    case PRODUCT_SUBMISSION_STATUS.NOT_STARTED:
      return {
        label,
        value: SUBMISSION_STATUS.NOT_STARTED,
      };
  }
};
