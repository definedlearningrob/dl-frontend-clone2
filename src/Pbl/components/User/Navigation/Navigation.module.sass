@import '@shared/stylesheets/config.sass'

$highlighted-section-height: 10px
$highlighted-section-margin: $m-md + $highlighted-section-height $m-xxs $m-xs

.highlightedSection
  margin-top: $m-md
  padding-left: $m-sm
  margin-bottom: $m-xs

.small
  height: 2px
  background-color: $color-neutral-700
  margin: $highlighted-section-margin

.highlightedSectionDivide
  +kicker()
  padding-left: 0
  margin: 0
  color: $color-neutral-700
