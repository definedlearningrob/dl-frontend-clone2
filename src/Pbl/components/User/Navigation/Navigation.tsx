import { useTranslation } from 'react-i18next';
import { isEmpty, some } from 'lodash-es';
import cx from 'classnames';

import { ReactComponent as Message } from '@dc/svg/messages.svg';

import useUserInfo from '@pbl/hooks/useUserInfo';
import { ReactComponent as Home } from '@pbl/svg/home_dashboard.svg';
import { ReactComponent as Projects } from '@pbl/svg/projects.svg';
import { THighlightedCatalogOption, TUserInfo } from '@pbl/graphql/user/queries/userInfo';
import { Roles } from '@pbl/resources/enums';

import { HighlightedCatalog } from '@shared/components/Sidebar/HighlightCatalog/highlightedCatalog';
import {
  SidebarNavItem,
  type SidebarNavItemProps,
} from '@shared/components/Sidebar/NavItem/NavItem';
import { ReactComponent as Educator } from '@shared/svg/educator.svg';
import { ReactComponent as Book } from '@shared/svg/book_opened.svg';
import { ReactComponent as ClassIcon } from '@shared/svg/class.svg';
import { ReactComponent as GoalsIcon } from '@shared/assets/icons/sport-cup.svg';
import MessagingModal from '@shared/components/Messaging/Modal/Modal';
import { REPORT_PATHS, RECEIVER_TYPES, DL_REPORTS } from '@shared/resources/constants';
import { useFeatureFlags } from '@shared/components/FeatureProvider';
import { useMessaging } from '@shared/hooks/useMessaging';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';
import { GOALS_REPORTS } from '@shared/resources/constants';

import style from './Navigation.module.sass';

export const Navigation = () => {
  const { t } = useTranslation();
  const { userInfo } = useUserInfo<TUserInfo>();
  const { messagingState, setMessagingState } = useMessaging();
  const { DL_ADMIN } = useFeatureFlags();
  const { isExpanded } = useNavigation();

  const canBrowseReports =
    userInfo.permissions.canBrowseReports &&
    some(
      userInfo.availableReportTypes,
      (reportType) => GOALS_REPORTS.includes(reportType) || DL_REPORTS.includes(reportType)
    );

  const isOnlyOneReport = userInfo.availableReportTypes.length === 1;

  const navItems: SidebarNavItemProps[] = [
    {
      id: 'home',
      icon: <Home />,
      path: '/',
      text: t('user.navigation.home'),
    },
    {
      id: 'reports',
      path: isOnlyOneReport
        ? `/reports/${REPORT_PATHS[userInfo.availableReportTypes[0]]}`
        : '/reports',
      icon: <GoalsIcon />,
      text: t('user.navigation.reports'),
      hidden: !canBrowseReports,
    },
    {
      id: 'myProjects',
      icon: <Projects />,
      path: '/my-projects',
      text: t('user.navigation.myProjects'),
    },
    {
      id: 'myClasses',
      icon: <ClassIcon />,
      path: '/my-classes',
      text: t('user.navigation.myClasses'),
    },
    {
      id: 'messaging',
      icon: <Message />,
      needsAttention: userInfo?.hasUnreadConversation,
      path: '/messages',
      text: t('navigation.messages'),
    },
    {
      id: 'dashboard',
      icon: <Educator />,
      path: '/admin-dashboard',
      text: t('user.navigation.dashboard'),
      hidden: !DL_ADMIN || ![Roles.ENTITY_ADMIN, Roles.SYSTEM_ADMIN].includes(userInfo.role),
    },
  ];

  const filteredNavItems = navItems.filter((item) => !item.hidden);
  const highlightedCatalogsData = userInfo?.highlightedCatalogs.map(
    (catalog: THighlightedCatalogOption) => ({
      id: catalog.id,
      text: catalog.name,
      icon: <Book />,
      path: `/my-catalogs/${catalog.id}`,
    })
  );

  const toggleMessagingModal = () => {
    setMessagingState({ ...messagingState, show: !messagingState.show });
  };

  const highlightedSectionClassnames = cx(style.highlightedSection, {
    [style.small]: !isExpanded,
  });

  return (
    <nav>
      {filteredNavItems.map((item: SidebarNavItemProps) => (
        <SidebarNavItem key={item.id} {...item} />
      ))}
      {messagingState.show && (
        <MessagingModal
          initialReceiver={messagingState.receiver}
          receiverType={RECEIVER_TYPES.STUDENT}
          toggleModal={toggleMessagingModal}
        />
      )}
      {!isEmpty(highlightedCatalogsData) && (
        <>
          <div className={highlightedSectionClassnames}>
            {isExpanded && (
              <p className={style.highlightedSectionDivide}>{t('user.navigation.featured')}</p>
            )}
          </div>
          {highlightedCatalogsData.map((item) => (
            <HighlightedCatalog key={item.id} catalog={item} />
          ))}
        </>
      )}
    </nav>
  );
};
