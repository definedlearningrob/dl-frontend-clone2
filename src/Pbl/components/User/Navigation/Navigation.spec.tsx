import { MockedProvider } from '@apollo/client/testing';
import { screen } from '@testing-library/dom';
import { ReportTypes } from '@graphql/shared/shared/types';

import { UserInfoProvider } from '@pbl/hooks/useUserInfo';
import DlUserInfo, { TUserInfo } from '@pbl/graphql/user/queries/userInfo';
import { renderWithRouterAndReduxProvider } from '@pbl/utils/test';

import { PresentationStateProvider } from '@shared/hooks/usePresentationState';
import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';
import { MessagingProvider } from '@shared/hooks/useMessaging';

import { Navigation } from './Navigation';

export const dlUserInfoMock = {
  request: {
    query: DlUserInfo,
    variables: {
      // username: undefined,
    },
  },
  result: {
    data: {
      userInfo: {
        availableTasksCount: 636,
        availableTracksCount: 57,
        availableReportTypes: [
          'CAREER_REVIEW_SURVEY',
          'ASSESSMENT',
          'COLLEGE_AND_FUTURE',
          'OPPORTUNITIES',
          'CAREER_PATHWAY',
        ],
        algoliaSearchKey: 'abc',
        currentSchoolYear: 2024,
        hasAccessToCareers: true,
        logoUrl: null,
        iconUrl: null,
        permissions: {
          canBrowseReports: true,
        },
        hasUnreadConversation: false,
        entities: {
          nodes: [
            {
              settings: {
                classManagementEnabled: true,
                schoolYearStartDate: {
                  day: 1,
                  month: 7,
                },
              },
              uuid: '11111111-1111-1111-1111-111111111111',
            },
          ],
        },
        highlightedCatalogs: [],
        email: 'alonzorockteach@cleverdemo.com',
        firstName: 'Alonzo',
        lastName: 'Rockteach',
        role: 'TEACHER',
        standardSets: [
          {
            id: '14',
            name: 'Arkansas',
            setId: 'AR',
          },
        ],
        username: 'cleverteacher5',
        uuid: '12345678-1234-1234-1234-123456789012',
        id: '1221256',
        definedLearningUuid: '12345678-1234-1234-1234-123456789012',
        ltiDetails: {
          isLti: false,
          isLtiSearch: false,
          ltiContextId: '12345678-1234-1234-1234-123456789012',
          ltiConsumerKey: 'abc',
          ltiResourceLinkId: '1234413',
        },
        welcomeMessage: 'hello!',
      },
    },
  },
};

describe('Navigation', () => {
  it('renders correctly', () => {
    const { container } = renderWithRouterAndReduxProvider(
      <MockedProvider mocks={[dlUserInfoMock]}>
        <MessagingProvider refreshUser={jest.fn()} userInfo={{} as TUserInfo}>
          <PresentationStateProvider>
            <NavigationContextProvider>
              <UserInfoProvider value={{ userInfo: { ...dlUserInfoMock.result.data.userInfo } }}>
                <Navigation />
              </UserInfoProvider>
            </NavigationContextProvider>
          </PresentationStateProvider>
        </MessagingProvider>
      </MockedProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it('render correctly while teacher has no access to reports', () => {
    const { container } = renderWithRouterAndReduxProvider(
      <MockedProvider mocks={[dlUserInfoMock]}>
        <MessagingProvider refreshUser={jest.fn()} userInfo={{} as TUserInfo}>
          <PresentationStateProvider>
            <NavigationContextProvider>
              <UserInfoProvider
                value={{
                  userInfo: {
                    ...dlUserInfoMock.result.data.userInfo,
                    permissions: { canBrowseReports: false },
                  },
                }}>
                <Navigation />
              </UserInfoProvider>
            </NavigationContextProvider>
          </PresentationStateProvider>
        </MessagingProvider>
      </MockedProvider>
    );

    const reportsLink = screen.queryByRole('link', { name: 'Goals & Reports' });

    expect(reportsLink).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('render correctly while there is only Career Exploration report available', async () => {
    const { container } = renderWithRouterAndReduxProvider(
      <MockedProvider mocks={[dlUserInfoMock]}>
        <MessagingProvider refreshUser={jest.fn()} userInfo={{} as TUserInfo}>
          <PresentationStateProvider>
            <NavigationContextProvider>
              <UserInfoProvider
                value={{
                  userInfo: {
                    ...dlUserInfoMock.result.data.userInfo,
                    canBrowseReports: false,
                    availableReportTypes: [ReportTypes.CAREER_PATHWAY],
                  },
                }}>
                <Navigation />
              </UserInfoProvider>
            </NavigationContextProvider>
          </PresentationStateProvider>
        </MessagingProvider>
      </MockedProvider>
    );

    const reportsLink = await screen.findByRole('link', { name: 'Goals & Reports' });

    expect(reportsLink).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
