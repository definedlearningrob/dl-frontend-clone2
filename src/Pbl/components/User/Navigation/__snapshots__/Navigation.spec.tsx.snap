// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation render correctly while teacher has no access to reports 1`] = `
<div>
  <nav>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          aria-current="page"
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item text-primary-500 bg-primary-200"
          data-testid="navigation-item-home"
          href="/"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Home
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-myProjects"
          href="/my-projects"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            My Projects
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-myClasses"
          href="/my-classes"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            My Classes
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-messaging"
          href="/messages"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Messages
          </span>
        </a>
      </span>
    </div>
  </nav>
</div>
`;

exports[`Navigation render correctly while there is only Career Exploration report available 1`] = `
<div>
  <nav>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          aria-current="page"
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item text-primary-500 bg-primary-200"
          data-testid="navigation-item-home"
          href="/"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Home
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-reports"
          href="/reports/career-exploration"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Goals & Reports
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-myProjects"
          href="/my-projects"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            My Projects
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-myClasses"
          href="/my-classes"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            My Classes
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-messaging"
          href="/messages"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Messages
          </span>
        </a>
      </span>
    </div>
  </nav>
</div>
`;

exports[`Navigation renders correctly 1`] = `
<div>
  <nav>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          aria-current="page"
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item text-primary-500 bg-primary-200"
          data-testid="navigation-item-home"
          href="/"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Home
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-reports"
          href="/reports"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Goals & Reports
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-myProjects"
          href="/my-projects"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            My Projects
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-myClasses"
          href="/my-classes"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            My Classes
          </span>
        </a>
      </span>
    </div>
    <div
      class="relative"
    >
      <span
        data-state="closed"
      >
        <a
          class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
          data-testid="navigation-item-messaging"
          href="/messages"
        >
          <div
            class="icon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="font-medium truncate text opacity-0"
          >
            Messages
          </span>
        </a>
      </span>
    </div>
  </nav>
</div>
`;
